core/src/main/java/org/apache/accumulo/core/constraints/NoDeleteConstraint.java

```diff
@@ -40,15 +47,20 @@ public String getViolationDescription(short violationCode) {
   }
 
   @Override
-  public List<Short> check(Environment env, Mutation mutation) {
+  public List<Short> check(Constraint.Environment env, Mutation mutation) {
     List<ColumnUpdate> updates = mutation.getUpdates();
     for (ColumnUpdate update : updates) {
       if (update.isDeleted()) {
         return Collections.singletonList((short) 1);
       }
     }
-
     return null;
   }
 
+  @Override
+  public List<Short> checkMutation(
+      org.apache.accumulo.core.data.constraints.Constraint.Environment env, Mutation mutation) {
+    return check((Constraint.Environment) env, mutation);
+  }
+
```

Comment:
Will the default method added to the old Constraint interface not work for these? It seems like these aren't necessary. If they are necessary, then it will be necessary for users as well, if they wrote their own constraints. It'd be nice if it wasn't necessary for them, and if so, it's not necessary for us, either.

Reply:
I had created the new method to get around this error:
<pre>
[ERROR] /home/mike/workspace/accumulo/core/src/test/java/org/apache/accumulo/core/constraints/NoDeleteConstraintTest.java:[47,18] reference to check is ambiguous
  both method check(org.apache.accumulo.core.data.constraints.Constraint.Environment,org.apache.accumulo.core.data.Mutation) in org.apache.accumulo.core.data.constraints.NoDeleteConstraint and method check(org.apache.accumulo.core.constraints.Constraint.Environment,org.apache.accumulo.core.data.Mutation) in org.apache.accumulo.core.constraints.NoDeleteConstraint match
</pre>

But I finally figured out I needed to make the `Environment` interface extend the new one as well. It looks like we are able to keep the same method with this change.

Reply:
I am not sure if this will be an issue for users or not. I am getting this spotbugs error in the Test I made for extending one of the constraints:
`
[ERROR] Medium: org.apache.accumulo.core.constraints.DeprecatedConstraintExtendTest$MinKeySizeConstraint.check(Constraint$Environment, Mutation) doesn't override method in superclass because parameter type org.apache.accumulo.core.constraints.Constraint$Environment doesn't match superclass parameter type org.apache.accumulo.core.data.constraints.Constraint$Environment [org.apache.accumulo.core.constraints.DeprecatedConstraintExtendTest$MinKeySizeConstraint] At DeprecatedConstraintExtendTest.java:[lines 77-90] NM_WRONG_PACKAGE
`